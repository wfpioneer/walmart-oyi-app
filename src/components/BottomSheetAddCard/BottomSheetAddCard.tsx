/* eslint-disable react/require-default-props */
import React from 'react';
import BottomSheetBaseCard from '../BottomSheetBaseCard/BottomSheetBaseCard';

interface AddCardProps {
  isManagerOption?: boolean;
  isVisible?: boolean;
  text: string;
  onPress: () => void;
}

const BottomSheetAddCard = (props: AddCardProps): JSX.Element | null => {
  const {
    isManagerOption, isVisible, text, onPress
  } = props;
  let image;
  if (isManagerOption) {
    image = require('../../assets/images/add_circle_dark_grey.png');
  } else {
    image = require('../../assets/images/add_circle_light_grey.png');
  }

  return isVisible ? (
    <BottomSheetBaseCard image={image} onPress={onPress} text={text} />
  ) : null;
};

export default BottomSheetAddCard;
