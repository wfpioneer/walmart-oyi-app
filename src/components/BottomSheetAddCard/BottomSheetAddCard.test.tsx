import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import BottomSheetAddCard from './BottomSheetAddCard';

describe('Add card render tests', () => {
  it('renders the bottom sheet add card, not visible', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BottomSheetAddCard
        onPress={() => {}}
        text="hello"
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders the bottom sheet add card, visible, non manager option', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BottomSheetAddCard
        onPress={() => {}}
        text="hello"
        isVisible
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('renders the bottom sheet add card, visible, manager option', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BottomSheetAddCard
        onPress={() => {}}
        text="hello"
        isVisible
        isManagerOption
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
