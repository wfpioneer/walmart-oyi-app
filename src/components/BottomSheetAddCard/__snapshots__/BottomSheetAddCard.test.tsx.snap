// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add card render tests renders the bottom sheet add card, not visible 1`] = `null`;

exports[`Add card render tests renders the bottom sheet add card, visible, manager option 1`] = `
<BottomSheetBaseCard
  image=""
  onPress={[Function]}
  text="hello"
/>
`;

exports[`Add card render tests renders the bottom sheet add card, visible, non manager option 1`] = `
<BottomSheetBaseCard
  image=""
  onPress={[Function]}
  text="hello"
/>
`;
